File: java/AntiXSS.java
import java.util.regex.Pattern;

public class AntiXSS {

    //region Anti xss
    public static String antiXSS(String input, boolean disableHtmlEncoding, String allowRegex) {
        try {
            if (!Pattern.matches(allowRegex, input)) {
                return "Dangerous Input";
            }
            input = disableHtmlEncoding ? input : HtmlUtility.encode(input);
            return input;
        } catch (Exception ex) {
            return "Error";
        }
    }

    public static String antiXSS(String input, boolean disableHtmlEncoding) {
        return antiXSS(input, disableHtmlEncoding, "^[a-zA-Z0-9]*");
    }

    public static String antiXSS(String input) {
        return antiXSS(input, false, "^[a-zA-Z0-9]*");
    }
    //endregion
}

File: CSharp/AntiXSS.cs
using System;
using System.Text;
using System.Text.RegularExpressions;

namespace PC.Framework
{
  public static class AntiXSS
  {
     #region *** Antix XSS ***

        public static string AntiXSS(string input, bool disableHtmlEncoding = false, string allowRegex = "^[a-zA-Z0-9, !@#$%^&*()_+=-]*")
        {
            try
            {
                if (!Regex.IsMatch(input, allowRegex))
                    return "Dangerous Input";
                input = disableHtmlEncoding ? input : HttpUtility.HtmlEncode(input);
                return input;
            }
            catch (Exception ex)
            {
                return "Error";
            }
        }

        #endregion
  }
}
